launch/launch/launch_description_sources/any_launch_file_utilities.py

```diff
@@ -39,14 +39,15 @@ def get_launch_description_from_any_launch_file(
     :raise `ValueError`: Invalid file. The file may not be a text file.
     """
     loaders = [get_launch_description_from_frontend_launch_file]
-    extension = os.path.splitext(launch_file_path)[1]
+    launch_file_name = os.path.basename(launch_file_path)
```

Comment:
is this line needed?

Reply:
the filename is used below for the call to `Parser.is_filename_valid()`, even if in practice (because we're only checking the end of the filename) using the file path wouldn't change anything